server/services/omnichannel-voip/service.ts

```diff
@@ -72,4 +155,156 @@ export class OmnichannelVoipService extends ServiceClass implements IOmnichannel
+			const roomId = await this.createVoipRoom(rid, name, agent, guest);
+			room = await this.voipRoom.findOneVoipRoomById(roomId);
+			newRoom = true;
+			this.logger.debug(`Room obtained for visitor ${guest._id} -> ${room?._id}`);
+		}
+		if (!room) {
+			this.logger.debug(`Visitor ${guest._id} trying to access another visitor's room`);
+			throw new Meteor.Error('cannot-access-room');
+		}
+		return {
+			result: {
+				room,
+				newRoom,
+			},
+		};
+	}
+
+	async findRoom(token: string, rid: string): Promise<IOmnichannelVoipServiceResult> {
+		const fields = {
+			t: 1,
+			departmentId: 1,
+			servedBy: 1,
+			open: 1,
+			v: 1,
+			ts: 1,
+		};
+		let room: IVoipRoom | null;
+		if (!rid) {
+			room = await this.voipRoom.findOneByVisitorToken(token, fields);
+		} else {
+			room = await this.voipRoom.findOneByIdAndVisitorToken(rid, token, fields);
+		}
+		return Promise.resolve({
+			result: room,
+		});
+	}
+
+	async closeRoom(visitor: ILivechatVisitor, room: IVoipRoom, /* comment: any,*/ options = {}): Promise<any> {
+		this.logger.debug(`Attempting to close room ${room._id}`);
+		if (!room || room.t !== 'v' || !room.open) {
+			return false;
+		}
+
+		const params = callbacks.run('livechat.beforeCloseRoom', { room, options });
+		const { extraData } = params;
+
+		const now = new Date();
+		const { _id: rid } = room;
+
+		const closeData = {
+			closedAt: now,
+			callDuration: now.getTime() / 1000,
+			...extraData,
+		};
+		this.logger.debug(`Room ${room._id} was closed at ${closeData.closedAt} (duration ${closeData.chatDuration})`);
+
+		/*
+		// We should be able to handle nearend and farend call end.
+		if (user) {
+			this.logger.debug(`Closing by user ${ user._id }`);
+			closeData.closer = 'user';
+			closeData.closedBy = {
+				_id: user._id,
+				username: user.username,
+			};
+		} else
+		*/
+		if (visitor) {
+			this.logger.debug(`Closing by visitor ${visitor._id}`);
+			closeData.closer = 'visitor';
+			closeData.closedBy = {
+				_id: visitor._id,
+				username: visitor.username,
+			};
+		}
+
+		this.voipRoom.closeByRoomId(rid, closeData);
+		// Retreive the closed room
+		/**
+		 * Note (Amol) How do we handle the code below
+		 */
+		/*
+		const closedRoom = await VoipRoom.findOneByIdOrName(rid, {});
+
+		this.logger.debug(`Sending closing message to room ${ room._id }`);
+		const message = {
+			t: 'livechat-close',
+			msg: comment,
+			groupable: false,
+			transcriptRequested: !!transcriptRequest,
+		};
+		sendMessage(visitor, message, closedRoom);
+
+
+		Meteor.defer(() => {
+			/**
+			 * @deprecated the `AppEvents.ILivechatRoomClosedHandler` event will be removed
+			 * in the next major version of the Apps-Engine
+
+			Apps.getBridges().getListenerBridge().livechatEvent(AppEvents.ILivechatRoomClosedHandler, room);
+			Apps.getBridges().getListenerBridge().livechatEvent(AppEvents.IPostLivechatRoomClosed, room);
+		});
+		callbacks.runAsync('livechat.closeRoom', room);
+		*/
+		return Promise.resolve({
```

Comment:
no need to resolve values inside async functions

Reply:
Fixed.